koku/masu/database/trino_sql/reporting_ocpazurecostlineitem_daily_summary.sql

```diff
@@ -285,7 +298,272 @@ GROUP BY coalesce(azure.date, azure.usagedatetime),
+            OR
+                (lower(ocp.csi_volume_handle) = lower(azure.resource_id))
+            )
+    WHERE ocp.source = {{ocp_source_uuid}}
+        AND ocp.year = {{year}}
+        AND lpad(ocp.month, 2, '0') = {{month}}
+        AND ocp.usage_start >= {{start_date}}
+        AND ocp.usage_start < date_add('day', 1, {{end_date}})
+        AND azure.ocp_source = {{ocp_source_uuid}}
+        AND azure.year = {{year}}
+        AND azure.month = {{month}}
+)
+SELECT
+    ocp_filtered.azure_partial_resource_id,
+    max(az_disk_capacity.capacity) as capacity,
+    date(coalesce(date, usagedatetime)) as usage_start,
+    {{ocp_source_uuid}} as ocp_source,
+    {{year}} as year,
+    {{month}} as month
+FROM azure_line_items as azure
+JOIN postgres.public.reporting_common_diskcapacity as az_disk_capacity
+    ON azure.metername LIKE '%' || az_disk_capacity.product_substring || ' %' -- space here is important to avoid partial matching
+    AND az_disk_capacity.provider_type = 'Azure'
+JOIN cte_ocp_filtered_resources as ocp_filtered
+    ON split_part(coalesce(nullif(azure.resourceid, ''), azure.instanceid), '/', 9) = ocp_filtered.azure_partial_resource_id
+WHERE coalesce(azure.date, azure.usagedatetime) >= TIMESTAMP '{{start_date | sqlsafe}}'
+    AND coalesce(azure.date, azure.usagedatetime) < date_add('day', 1, TIMESTAMP '{{end_date | sqlsafe}}')
+    AND coalesce(nullif(azure.servicename, ''), azure.metercategory) LIKE '%Storage%'
+    AND coalesce(nullif(azure.resourceid, ''), azure.instanceid) LIKE '%%Microsoft.Compute/disks/%%'
+    AND azure.year = {{year}}
+    AND azure.month = {{month}}
+GROUP BY ocp_filtered.azure_partial_resource_id, date(coalesce(date, usagedatetime))
+{% endif %}
+;
+
+{% if unattributed_storage %}
+-- resource_id matching
+-- Storage disk resources:
+-- (PV’s Capacity) / Disk Capacity * Cost of Disk
+-- PV without PVCs are Unattributed Storage
+INSERT INTO hive.{{schema | sqlsafe}}.reporting_ocpazurecostlineitem_project_daily_summary_temp (
+    azure_uuid,
+    cluster_id,
+    cluster_alias,
+    data_source,
+    namespace,
+    node,
+    persistentvolumeclaim,
+    persistentvolume,
+    storageclass,
+    resource_id,
+    usage_start,
+    usage_end,
+    service_name,
+    instance_type,
+    subscription_guid,
+    subscription_name,
+    resource_location,
+    unit_of_measure,
+    usage_quantity,
+    currency,
+    pretax_cost,
+    markup_cost,
+    pod_labels,
+    volume_labels,
+    tags,
+    resource_id_matched,
+    cost_category_id,
+    ocp_source,
+    year,
+    month
+)
+SELECT azure.uuid as azure_uuid,
+    max(ocp.cluster_id) as cluster_id,
+    max(ocp.cluster_alias) as cluster_alias,
+    'Storage' as data_source,
+    CASE
+        WHEN max(persistentvolumeclaim) = ''
+            THEN 'Storage unattributed'
+        ELSE max(namespace)
+    END as namespace,
+    max(ocp.node) as node,
```

Comment:
(question) What is the result of `max(ocp.node)` here for claimless PVs? Below (in lines 461-564), when inserting for `Storage unattributed`, the node is null, but here it seems like the node would be not null. Is that accurate?

Reply:
The node column is null in the ocp daily summary table for the claimless persistent volume. So, it would be null here too:
```
|       node        | persistentvolumeclaim |  persistentvolume   | storageclass |       resource_id        |
| NULL              |                       | pv-volume_claimless | sc_disk_one  | abc_pv-volume_claimless  |
```